src/ripple/app/ledger/impl/LedgerMaster.cpp

```diff
@@ -1478,12 +1465,9 @@ LedgerMaster::getPropertySource ()
 void
 LedgerMaster::clearPriorLedgers (LedgerIndex seq)
 {
-    ScopedLockType sl (mCompleteLock);
-    for (LedgerIndex i = mCompleteLedgers.getFirst(); i < seq; ++i)
-    {
-        if (haveLedger (i))
-            clearLedger (i);
-    }
+    ScopedLockType sl(mCompleteLock);
+    if (seq > 0)
+        mCompleteLedgers.erase(range(0u, seq - 1));
```

Comment:
I *think* so, but I think the design of `LedgerMaster` doesn't enforce it well, so I was wary to change it.